src/Controller/ApiV1RedirectController.php

```diff
@@ -6,13 +6,20 @@
 
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Attribute\Route;
 
 class ApiV1RedirectController extends AbstractController
 {
     #[Route('/v1/{endpoint}', name: 'app_api_v1_redirect', requirements: ['endpoint' => '.+'], defaults: ['endpoint' => null], methods: ['GET'])]
     public function index(string $endpoint): RedirectResponse
     {
-        return $this->redirect('/v2/'.$endpoint, \Symfony\Component\HttpFoundation\Response::HTTP_MOVED_PERMANENTLY);
+        return $this->redirect('/v2/'.$endpoint, Response::HTTP_MOVED_PERMANENTLY);
```

Comment:
We can consolidate this to

```
<?php

declare(strict_types=1);

namespace App\Controller;

use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\RedirectResponse;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\Routing\Attribute\Route;

class ApiV1RedirectController extends AbstractController
{
    #[Route('/v1/{endpoint}', name: 'app_api_v1_redirect', requirements: ['endpoint' => '.+'], defaults: ['endpoint' => null])]
    public function index(string $endpoint): RedirectResponse
    {
        return $this->redirect('/v2/'.$endpoint, Response::HTTP_PERMANENTLY_REDIRECT);
    }
}
```
`HTTP_PERMANENTLY_REDIRECT` ([HTTP 308](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/308)):


> The request method and the body will not be altered, whereas [301](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/301) may incorrectly sometimes be changed to a [GET](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET) method.

By not specifying any methods the route applies to all:
```
$ idc bin/console debug:router app_api_v1_redirect 
+--------------+--------------------------------------------------------------+
| Property     | Value                                                        |
+--------------+--------------------------------------------------------------+
| Route Name   | app_api_v1_redirect                                          |
| Path         | /v1/{endpoint}                                               |
| Path Regex   | {^/v1(?:/(?P<endpoint>.+))?$}sDu                             |
| Host         | ANY                                                          |
| Host Regex   |                                                              |
| Scheme       | ANY                                                          |
| Method       | ANY                                                          |
| Requirements | endpoint: .+                                                 |
| Class        | Symfony\Component\Routing\Route                              |
| Defaults     | _controller: App\Controller\ApiV1RedirectController::index() |
|              | endpoint: NULL                                               |
| Options      | compiler_class: Symfony\Component\Routing\RouteCompiler      |
|              | utf8: true                                                   |
+--------------+--------------------------------------------------------------+
```